### PR TITLE
Add support for customer bank accounts in paymentsources

### PR DESCRIPTION
**Pull request by wbronitsky**

### CODE REVIEW

**Comment by wbronitsky** [comment]
@brandur @remistr Updated support for customer bank account sources. `#Update` was left out as `metadata` needs to be added as a thing first

File: bankaccount.go [hunk 1]
```diff
@@ -24,7 +24,7 @@ type BankAccountParams struct {
 
 	// Information on an external account to reference. Only used if `Token`
 	// is not provided.
-	Account, Country, Currency, Routing string
+	Account, Country, Currency, Routing, AccountHolderType, Name string
```
Comment (brandur) [inline bankaccount.go]: Nit, but mind keeping this list in alphabetical order since it's that way already Thanks!

File: bankaccount.go [hunk 2]
```diff
@@ -38,17 +38,19 @@ type BankAccountListParams struct {
...
-	LastFour    string            `json:"last4"`
-	Fingerprint string            `json:"fingerprint"`
-	Status      BankAccountStatus `json:"status"`
-	Routing     string            `json:"routing_number"`
-	Deleted     bool              `json:"deleted"`
-	Customer    *Customer         `json:"customer"`
+	ID          			string            `json:"id"`
+	BankName    			string        		`json:"bank_name"`
+	Name        			string						`json:"name"`
+	AccountHolderType string						`json:"account_holder_type"`
```
Comment (brandur) [inline bankaccount.go]: I think Remi got this on another commit already, but looks like there was some whitespace issues here.

File: paymentsource/client_test.go [hunk 3]
```diff
@@ -5,6 +5,8 @@ import (
 
 	stripe "github.com/stripe/stripe-go"
 	"github.com/stripe/stripe-go/card"
+	"github.com/stripe/stripe-go/bankaccount"
+	"github.com/stripe/stripe-go/token"
```
Comment (brandur) [inline paymentsource/client_test.go]: Actually, would you mind trying to run the changes through `gofmt`? It'll automatically re-order some of the imports like this one.

**Comment by brandur** [comment]
@wbronitsky Awesome! Thanks a lot for the pull and the comprehensive test suite.

I left a few minor formatting comments above, but this looks good to me. Could we just get your confirmation as well that you've run the new tests locally and seen them pass? Unfortunately this project's CI doesn't currently run most of the suite. Thanks!

**Comment by wbronitsky** [comment]
@brandur tests do pass locally, although there are for sure other broken tests in the suite.

File: paymentsource/client_test.go [hunk 4]
```diff
@@ -154,54 +274,58 @@ func TestSourceUpdate(t *testing.T) {
...
 	}
 
-	target := source.BankAccount
+	target := src.BankAccount
 
 	if target.Status != bankaccount.VerifiedAccount {
 		t.Errorf("Status (%q) does not match expected (%q) ", target.Status, bankaccount.VerifiedAccount)
 	}
-
-	customer.Del(cust.ID)
```
Comment (brandur) [inline paymentsource/client_test.go]: @wbronitsky Any reason to pull this `customer.Del` out? It's a little crazy, but this is actually how this binding's test suite cleans itself up (or tries to anyway).

File: paymentsource/client_test.go [hunk 5]
```diff
@@ -154,54 +274,58 @@ func TestSourceUpdate(t *testing.T) {
...
 
 	verifyParams := &stripe.SourceVerifyParams{
-		Customer: cust.ID,
-		Amounts:  amounts,
+		Customer: customer.ID,
+		Amounts: amounts,
 	}
 
-	source, err := Verify(cust.DefaultSource.ID, verifyParams)
+	src, err := Verify(source.ID, verifyParams)
```
Comment (brandur) [inline paymentsource/client_test.go]: Mind if we just use `sourceVerified` instead of `src` here instead?

**Comment by brandur** [comment]
@wbronitsky Sorry, lost track of this one. I left a few other minor comments, but let's try to get this in today.

File: bankaccount.go [hunk 6]
```diff
@@ -38,17 +38,19 @@ type BankAccountListParams struct {
...
-	Default     bool              `json:"default_for_currency"`
-	LastFour    string            `json:"last4"`
-	Fingerprint string            `json:"fingerprint"`
-	Status      BankAccountStatus `json:"status"`
-	Routing     string            `json:"routing_number"`
-	Deleted     bool              `json:"deleted"`
-	Customer    *Customer         `json:"customer"`
+	ID                string            `json:"id"`
+	Name              string            `json:"bank_name"`
+	AccountHolderName string						`json:"account_holder_name"`
```
Comment (brandur) [inline bankaccount.go]: @wbronitsky Mind just fixing the whitespace here? I'd try just a `gofmt -w`.

**Comment by brandur** [comment]
Let's get that whitespace fixed, but looks good to me!

**Comment by brandur** [comment]
Released as 11.3.0.

**Pull request merged**